scenes/vehicle/car_audio_controller.gd

```diff
@@ -0,0 +1,179 @@
+	car_audio_player_engine_on.play()
+	
+	car_audio_player_engine_off.stream = stream_engine_main_off
+	car_audio_player_engine_off.volume_db = -40
+	car_audio_player_engine_off.play()
+
+	car_audio_player_plop.stream = stream_plop
+	
+	car_audio_player_wind.stream = stream_drag_wind
+	
+	car_audio_player_brakes.stream = stream_brakes
+	
+	car_audio_player_gear_change.stream = stream_gear_change
+
+	car_audio_player_wind.volume_db = -50
+	car_audio_player_wind.play()
+	
+	
+	high_pass_filter = AudioServer.get_bus_effect(1, 0) as AudioEffectHighPassFilter
+	low_pass_filter = AudioServer.get_bus_effect(1, 1) as AudioEffectLowPassFilter
+
+
+func _process(delta):
+	#calculate db of drag wind
+	var car_linear_velocity_ms = car_ref.linear_velocity.length()
+	car_audio_player_wind.volume_db = -50 + (car_linear_velocity_ms * 0.7)
+	
+	rpms.push_front(car_ref.rpm)
+	if rpms.size() > rpm_interpolation_frames:
+		rpms.remove_at(rpms.size() - 1)
+	rpm_average = 0.0
+	for rpm in rpms:
+		rpm_average += rpm
+	rpm_average = rpm_average / rpms.size()
+	#calculate engine db based on rpm ,rpm_curve and blend it dependent on throttle pos
+	var rpm_normalized = (car_ref.rpm - idle_rpm) / (rev_limit_rpm - idle_rpm)
+	car_audio_player_engine_on.volume_db = (db_engine_on - 20) + clamp((20 * db_blend_curve_on.sample_baked(car_ref.throttle_input)), 0, 20) + (db_rpm_curve.sample_baked(rpm_normalized))
+	car_audio_player_engine_off.volume_db = (db_engine_off - 20) + (20 * db_blend_curve_off.sample_baked(car_ref.throttle_input)) +(db_rpm_curve.sample_baked(rpm_normalized))
+
+	# calculate pitch from car rpm
+	var pitch_out = clamp(rpm_average * 0.001, pitch_min, pitch_max)
+	
+	if filter_off_throttle:
+		high_pass_filter.cutoff_hz = clamp(20 + (db_blend_curve_off.sample_baked(car_ref.throttle_input) * (high_pass_cutoff_freq - 20)), 20 , 500)
+		low_pass_filter.cutoff_hz = 20500 - (db_blend_curve_off.sample_baked(car_ref.throttle_input) * (20500 - low_pass_cutoff_freq))
+	elif filter_on_throttle:
+		high_pass_filter.cutoff_hz = clamp(20 + (db_blend_curve_on.sample_baked(car_ref.throttle_input) * (high_pass_cutoff_freq - 20)), 20 , 500)
+		low_pass_filter.cutoff_hz = 20500 - (db_blend_curve_on.sample_baked(car_ref.throttle_input) * (20500 - low_pass_cutoff_freq))
+	else :
+		high_pass_filter.cutoff_hz = 20
+		low_pass_filter.cutoff_hz = 20500
+	print("cutoff highpass= " + str(high_pass_filter.cutoff_hz))
+
+	#fade in and out the brake disc sample 
+	if car_ref.brake_input > 0.5 and car_linear_velocity_ms > 5:
+		timer_brakes += delta
+		if timer_brakes > 0.1:
+			if !car_audio_player_brakes.playing:
+				car_audio_player_brakes.play()
+			if car_audio_player_brakes.volume_db < db_brake_noise:
+				car_audio_player_brakes.volume_db += delta * (50.0 / brake_noise_fade_in_time)
+				car_audio_player_brakes.pitch_scale = brake_noise_pitch
+	else :
+		timer_brakes = 0.0
+		if car_audio_player_brakes.volume_db > -50:
+			car_audio_player_brakes.volume_db -= delta * 50
```

Comment:
Do the brakes stop making noise if tires lock up? Hard to say sometimes if tires lock up since there's no tire squeal yet.

Reply:
when the tires have the sound logic on it i can rework it a bit. now it make noise even when the disc is locked...

the tires should manage the squeak for them selfs i think and maybe have a bool loked, or a lockage % value

should i implement the tire squeal on the tires or will you?

Reply:
Yea i think we could make bool for brakes_locked which is true if slip_ratio * sign(z_vel) == -1/1 else false. IIRC when going forward and brakes lock the slip ratio is 1 and -1 when doing the same but backwards. This needs to be confirmed though. Then use the break sound as we do now but stop playing if brakes_locked. 

The more i think about how to handle the tire sounds i am starting to lean towards the sound controller actually handling it. The reason is it already handles **_all_** other car sounds, so it would make sense for it to handle tire sounds as well.  We could for example have 4 audio stream nodes for each tire and set their position in the ready method to be its corresponding wheels position. Then in the _process method play the squeal sound if abs(slip) > peak_slip and change its pitch and volume based off of how much its slipping. 

Its up to you if you want to implement the tire sounds. If you don't, no problem i can also implement them.

Reply:
ok i will give it a try, should not be too complicated to implement it.